app/gosns/gosns.go

```diff
@@ -328,10 +329,10 @@ func Publish(w http.ResponseWriter, req *http.Request) {
 						msg.MessageBody = m
 					} else {
 						msg.MessageAttributes = messageAttributes
+						msg.MD5OfMessageAttributes = gosqs.HashAttributes(messageAttributes)
```

Comment:
HashAttributes should be moved to a common package and imported from gosqs and gosns to avoid any possibilities in the future of circular reference.  Any unit tests should be moved as well.